docs/usage/docker.md

```diff
@@ -206,6 +206,48 @@ module.exports = {
 };
 ```
 
+#### Google Container Registry
+
+Assume you are running Google Cloud a personal GitLab CI and storing the Docker images in the Google Container Registry (GCR).
+In this scenario the GCR requires token based authentication for everything and thus you must make two additional things happen:
+
+1. Get access to your token
+1. Make sure Renovate gets the token to read the Docker registry.
+
+_This documentation only gives **a few hints** on **a possible way** to achieve this in this scenario_
+
+You need a Renovate Docker image that includes the Google Cloud SDK which you have to build yourself.
+A rough sketch of what the Dockerfile to build such a custom Renovate image can look like.
+
+```Dockerfile
+FROM renovate/renovate:12.34.56
+# Include the "Docker tip" which you can find here https://cloud.google.com/sdk/docs/install
+# under "Installation" for "Debian/Ubuntu"
+RUN ...
+```
+
+In the `renovate.json` you can instruct Renovate to use a secret as the token to use when connecting the specified docker registry:
+
+```js
+{
+  "hostRules": [
+    {
+      "matchHost": "eu.gcr.io",
+      "token": "{{ secrets.GOOGLE_ACCESS_TOKEN }}"
+    }
+  ]
+}
+```
+
+You do not want to persist this token anywhere.
+One way to provide this token to Renovate is by creating a temporary file with the token from within the `.gitlab-ci.yml` using something like this:
+
+```yaml
+script:
+  - 'echo "module.exports = { secrets: { GOOGLE_ACCESS_TOKEN: ''"$(gcloud auth print-access-token)"'' } };" > config.js'
+  - renovate $RENOVATE_EXTRA_FLAGS
```

Comment:
I assume this code prints the token as plaintext into the `config.js` file? Don't you "leak" the token into each Docker container you start up this way?

Reply:
Inside the docker container is always has access to the token by running `gcloud auth print-access-token`.
The problem I'm still having is actually getting Renovate to use this token.
Current assessment is two fold: 
- I do not understand how the secrets is meant to be used.
- Renovate did not expect the need to have shortlived token.

So this part of the documentation will change before I take of the "Draft" label because this simply does not work.